sympy/assumptions/facts.py

```diff
@@ -34,71 +36,138 @@ def get_composite_predicates():
 
 @cacheit
 def get_known_facts():
+    """
+    Facts between unary predicates.
+
+    Returns
+    =======
+
+    x : Placeholder symbol for unary facts.
+
+    fact : Known facts in conjugated normal form.
+
+    """
     # We build the facts starting with primitive predicates.
     # DO NOT include the predicates in get_composite_predicates()'s keys here!
-    return And(
-
+    x = Symbol('x')
+    fact = And(
         # primitive predicates exclude each other
-        Implies(Q.negative_infinite, ~Q.positive_infinite),
-        Implies(Q.negative, ~Q.zero & ~Q.positive),
-        Implies(Q.positive, ~Q.zero),
+        Implies(Q.negative_infinite(x), ~Q.positive_infinite(x)),
+        Implies(Q.negative(x), ~Q.zero(x) & ~Q.positive(x)),
+        Implies(Q.positive(x), ~Q.zero(x)),
```

Comment:
All of these can be collected together as e.g.
```python
Q.negative_infinite(x) => ~(Q.negative(x) | Q.zero(x) | Q.positive(x) | Q.positive_infinite(x))
Q.negative(x) => ~(...)
...
```

Reply:
I think `Not(Q.negative_infinite(x) & Q.positive_infinite(x))` will be enough instead of `Q.negative_infinite(x) >> ~(Q.negative(x) | Q.zero(x) | Q.positive(x) | Q.positive_infinite(x))`, since finite and and infinite predicate reject each other. Would you take a look at how I changed this?

Reply:
To be clear; `Q.negative` already excludes `Q.negative_infinite` and `Q.positive_infinite` because
1. `Q.negative` implies `Q.finite`
2. Q.negative_infinite` implies `Q.infinite`
3. `Q.finite` and `Q.infinite` are mutually exclusive

Reply:
I think what you have is fine and I as @asmeurer says I don't think it is good to make non refactoring changes in a refactoring PR. It makes it messy in future with git bisect etc.

My point though (for another PR) is that there are many situations in the facts where we have a predicate that is equivalent to an or of a bunch mutually exclusive predicates like:
```
Equivalent(A, B | C)
Not(B & C)
```
With more variables that is
```
Equivalent(A, B | C | D | E)
Not(B & C)
Not(B & D)
Not(B & E)
Not(C & D)
Not(C & E)
Not(D & E)
```
It would be clearer and more compact if there was a way of saying that some set of predicates is mutually exclusive like:
```
Exclusive(B, C, D, E)
Equivalent(A, B | C | D | E)
```
Perhaps those could be combined like:
```
EquivalentExactlyOne(A, [B, C, D, E])
```
but maybe that's too much. You could use that like:
```
EquivalentExactlyOne(Q.complex(x), [Q.algebraic(x), Q.transcendental(x)])
EquivalentExactlyOne(Q.real(x), [Q.rational(x), Q.irrational(x)])
EquivalentExactlyOne(Q.integer(x), [Q.even(x), Q.odd(x)])

EquivalentExactlyOne(Q.extended_real(x), [
    Q.negative_infinite(x), Q.negative(x), Q.zero(x), Q.positive(x), Q.positive_infinite(x)])
```

Note that n-ary `Xor` doesn't work like `Exclusive`:
```
In [2]: Xor(True, True, True)
Out[2]: True
```

Reply:
The lines are modified anyway by parametrization, so I'll leave them as they are in 10ee1384279f3b903aab0ac17e4b178d4821b52d.

Reply:
I prefer `Not(A & B)` than `Implies(A, ~B)` for its symmetry, so I modified other facts this way and squashed the commits to make the history clean. I'll mark this conversation as resolved.